apps/backend/src/modules/user/user.controller.ts

```diff
@@ -0,0 +1,175 @@
+import {
+  Controller,
+  Get,
+  Post,
+  Body,
+  Patch,
+  HttpStatus,
+  UseGuards,
+  UploadedFiles,
+  Param,
+} from '@nestjs/common';
+import { UserService } from './user.service';
+import {
+  ApiBearerAuth,
+  ApiCreatedResponse,
+  ApiOperation,
+  ApiResponse,
+  ApiTags,
+} from '@nestjs/swagger';
+import { CognitoUserAttributes, User } from '@wagademy/types';
+import { CognitoUserGuard } from '../../infra';
+import { ApiFiles, CognitoUser, DBUser } from '../../shared/decorators';
+import {
+  CreateProfileEntity,
+  CreateUserResponseEntity,
+  FindProfileEntity,
+  RetrieveSelfResponseEntity,
+  UpdateProfileEntity,
+  UpdateUserResponseEntity,
+} from './entities';
+import {
+  CreateProfileDto,
+  CreateUserDto,
+  UpdateProfileDto,
+  UpdateUserDto,
+} from './dto';
+import { MongoIdDto } from '../../shared/dtos';
+
+@ApiTags('User')
+@Controller('user')
+export class UserController {
+  constructor(private readonly userService: UserService) {}
+
+  @Post()
+  @ApiBearerAuth()
+  @UseGuards(CognitoUserGuard)
+  @ApiOperation({
+    summary: 'Create a new user',
+    description: 'Creates a new user with the given cognito user.',
+  })
+  @ApiCreatedResponse({
+    type: CreateUserResponseEntity,
+    status: HttpStatus.CREATED,
+    description: 'User successfully created.',
+  })
+  create(
+    @CognitoUser()
+    { sub: idRefAuth, email }: CognitoUserAttributes,
+    @Body() { name, accountType }: CreateUserDto
```

Comment:
The profile picture, which the user can provide during sign up, is missing

Reply:
Unless you're considering making two API calls, one for 'create user' and another for 'create profile'

Reply:
This will be removed the picture in the signup